src/main.rs

```diff
@@ -1649,6 +1657,76 @@ fn cmd_regenerate_audit_as(
     Ok(())
 }
 
+fn cmd_renew(out: &Arc<dyn Out>, cfg: &Config, sub_args: &RenewArgs) -> Result<(), miette::Report> {
+    trace!("renewing wildcard audits");
+    let mut store = Store::acquire_offline(cfg)?;
+    do_cmd_renew(out, &mut store, sub_args);
+    store.commit()?;
+    Ok(())
+}
+
+fn do_cmd_renew(out: &Arc<dyn Out>, store: &mut Store, sub_args: &RenewArgs) {
+    assert!(sub_args.expiring ^ sub_args.crate_name.is_some());
+
+    let today = <chrono::DateTime<chrono::Utc>>::from(SystemTime::now()).date_naive();
+    let new_end_date = today + chrono::Months::new(12);
+
+    let renew_entry = |entry: &mut WildcardEntry| {
+        entry.end = new_end_date.into();
+    };
+
+    if let Some(name) = &sub_args.crate_name {
+        match store.audits.wildcard_audits.get_mut(name) {
+            Some(audits) => {
+                audits.iter_mut().for_each(renew_entry);
+                writeln!(
+                    out,
+                    "Updated wildcard audits for {name} to expire on {new_end_date}"
+                );
+            }
+            None => {
+                warn!("ran `renew {name}`, but there are no wildcard audits for the crate");
+            }
+        }
+    } else {
+        // Find and update all expiring crates.
+        assert!(sub_args.expiring);
+        let expire_date = today + *WILDCARD_AUDIT_EXPIRATION_DURATION;
+
+        let mut updated = Vec::new();
+        for (name, entries) in store.audits.wildcard_audits.iter_mut() {
+            let any = entries
+                .iter_mut()
+                .filter(|e| e.should_suggest_renewal(expire_date))
+                .any(|e| {
+                    renew_entry(e);
+                    true
+                });
```

Comment:
Why the `.any` here? I think this behaviour for short-circuiting and only updating a single entry is unintuitive, as it's also different behaviour than is used for the explicit crate case (where every wildcard entry will be renewed).

Could you make this an explicit loop instead rather than using a side-effectful `.any` condition?

Reply:
Sure. It was originally adapted from a `.for_each`, so it became awkward.